calicoctl/commands/get.go

```diff
@@ -66,53 +64,50 @@ Options:
 		return nil
 	}
 
-	cmd := get{}
-	results := executeConfigCommand(parsedArgs, cmd)
-	glog.V(2).Infof("results: %+v", results)
-
-	if results.err != nil {
-		fmt.Printf("Error getting resources: %v\n", results.err)
-		return err
+	var rp resourcePrinter
+	output := parsedArgs["--output"].(string)
+	switch output {
+	case "yaml":
+		rp = resourcePrinterYAML{}
+	case "json":
+		rp = resourcePrinterJSON{}
+	case "ps":
+		rp = resourcePrinterTable{wide: false}
+	case "wide":
+		rp = resourcePrinterTable{wide: true}
+	default:
+		// Output format may be a key=value pair, so split on "=" to find out.  Pull
+		// out the key and value, and split the value by "," as some options allow
+		// a multiple-valued value.
+		outputParms := strings.SplitN(output, "=", 2)
+		if len(outputParms) == 2 {
+			outputKey := outputParms[0]
+			outputValue := outputParms[1]
+			outputValues := strings.Split(outputValue, ",")
+			if len(outputParms) == 2 {
+				switch outputKey {
+				case "go-template":
+					rp = resourcePrinterTemplate{template: outputValue}
```

Comment:
Extra stuff will not be ignored!  For the options where we only expect a single value, then the single value is passed in, commas and all.

For the template-filename we will error if there is a comma in the name - because that won't be valid.

In the case of a template string, that may validly have a comma in it.  At the moment only the custom-columns uses the outputValues so I could move it into there if it's bugging you, but I kept it outside because I think it looked a little neater and we might end up adding other options that do have a comma separated list of things.